### PR TITLE
fix: node version check now uses process.versions.node

### LINKED ISSUE #447: Version number check fails on Bun
I decided to give Bun a try by running the Mocha test suite of one of my packages.  The very first failure was in `yargs-parser` like so:

```
$ bun run ./node_modules/mocha/bin/mocha.js test-partials.js test-parse.js test-process.js 
986 |     ? Number(process.env.YARGS_MIN_NODE_VERSION)
987 |     : 10;
988 | if (process && process.version) {
989 |     const major = Number(process.version.match(/v([^.]+)/)[1]);
990 |     if (major < minNodeVersion) {
991 |         throw Error(`yargs parser supports a minimum Node.js version of ${minNodeVersion}. Read our version support policy: https://github.com/yargs/yargs-parser#supported-nodejs-versions`);
                  ^
 error: yargs parser supports a minimum Node.js version of 10. Read our version support policy: https://github.com/yargs/yargs-parser#supported-nodejs-versions
      at /home/david/Projects/akasharender/mahabhuta/test/node_modules/yargs-parser/build/index.cjs:991:14
      at bun:wrap:1:16354
      at /home/david/Projects/akasharender/mahabhuta/test/node_modules/mocha/lib/cli/options.js:12:28
      at bun:wrap:1:16354
      at /home/david/Projects/akasharender/mahabhuta/test/node_modules/mocha/bin/mocha.js:13:30
```

Clearly, Bun isn't going to give a value for `process.version` that is compatible with Node.js versions.  With this script

```js
console.log(process.version);

console.log(process);
```

I get this output:

```
$ bun run pv.js 
v0.1.2
{ ... }
```

For reference: https://github.com/Jarred-Sumner/bun/issues/462

### PR DESCRIPTION
**Pull request by paperdave**
Closes #447 by using `process.versions.node` for testing the version. The only difference between these values is the `v` prefix, so it's not as trivial as swapping a variable. Also, just in case `process.versions` is somehow not set, this falls back to previous behavior.

With this change in place, Bun should be able to import yargs.

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -19,8 +19,9 @@ import { readFileSync } from 'fs'
 const minNodeVersion = (process && process.env && process.env.YARGS_MIN_NODE_VERSION)
   ? Number(process.env.YARGS_MIN_NODE_VERSION)
   : 12
-if (process && process.version) {
-  const major = Number(process.version.match(/v([^.]+)/)![1])
+const nodeVersion = process && (process.versions ? process.versions.node : process.version.slice(1))
```
Comment (bcoe) [inline lib/index.ts:22]: Could you do:

```
const nodeVersion = process?.versions?.node && process.version.slice(1);
```

To simplify the logic slightly?

**Review by bcoe — CHANGES_REQUESTED** [review]
Fix seem reasonable to me, left a small nit.

File: lib/index.ts (hunk 1 above)
Comment (davecaruso) [reply lib/index.ts:22]: this should be an `||` but yes the logic could be simplified. i wasn't sure if i was able to use `?.` in the code.

File: lib/index.ts [hunk 2]
```diff
@@ -19,8 +19,9 @@ import { readFileSync } from 'fs'
 const minNodeVersion = (process && process.env && process.env.YARGS_MIN_NODE_VERSION)
   ? Number(process.env.YARGS_MIN_NODE_VERSION)
   : 12
-if (process && process.version) {
-  const major = Number(process.version.match(/v([^.]+)/)![1])
+const nodeVersion = process?.versions?.node || process?.version?.slice(1)
```
Comment (bcoe) [inline lib/index.ts:22]:
```suggestion
const nodeVersion = process?.versions?.node ?? process?.version?.slice(1)
```

**Comment by bcoe** [comment]
@davecaruso I believe there's been a small dip in coverage because of the newline added, please feel free to update the file here:

https://github.com/yargs/yargs-parser/blob/main/.nycrc

On your branch, to bring the threshold down to 97.

**Review by bcoe — APPROVED** [review]

**Comment by bcoe** [comment]
@paperdave thank you for the contribution 👌

**Pull request merged**